src/ConfigCatClientCache.ts

```diff
@@ -0,0 +1,52 @@
+import { IConfigCatKernel } from "./index";
+import { ConfigCatClientOptions } from "./ConfigCatClientOptions";
+import { ConfigCatClient } from "./ConfigCatClient";
+
+export class ConfigCatClientCache {
+    private instances: Record<string, [WeakRef<ConfigCatClient>, object]> = {};
+
+    public getOrCreate(options: ConfigCatClientOptions, configCatKernel: IConfigCatKernel): [ConfigCatClient, boolean] {
+        let instance: ConfigCatClient | undefined;
+
+        let cachedInstance = this.instances[options.apiKey];
+        if (cachedInstance) {
+            const [weakRef] = cachedInstance;
+            instance = weakRef.deref();
+            if (instance) {
+                return [instance, true];
+            }
+        }
+
+        const token = {};
+        instance = new ConfigCatClient(options, configCatKernel, token);
+        this.instances[options.apiKey] = cachedInstance = [new WeakRef(instance), token];
```

Comment:
I can not see why did you set the `cachedInstance`...can you explain it?

Reply:
Neither can I... :) Looks like an accident which happened when I refactored this method.